crates/gen/src/types/struct.rs

```diff
@@ -69,6 +71,11 @@ impl Struct {
                     self as *mut Self::Abi
                 }
             }
+            #extensions
         }
     }
+
+    fn add_extensions(&self) -> TokenStream {
+        crate::numerics::generate_struct_extensions(self)
```

Comment:
Maybe just call this function above rather than wrapping it.

Reply:
Sounds good to me, it can always be introduced later if we have more types of extensions.